### PR TITLE
Add fixes for menu from xash3d repo.

### PR DESCRIPTION
**Pull request by nekonomicon**

### CODE REVIEW

File: mainui/menu_creategame.cpp [hunk 1]
```diff
@@ -94,6 +94,9 @@ static void UI_CreateGame_Begin( void )
 	if( CVAR_GET_FLOAT( "host_serverstate" ) && CVAR_GET_FLOAT( "maxplayers" ) == 1 )
 		HOST_ENDGAME( "end of the game" );
 
+	if( atoi( uiCreateGame.maxClients.buffer ) > 32 )
```
Comment (a1batross) [inline mainui/menu_creategame.cpp]: twice atoi() call?

Also, it will be bounded by engine anyway, from 1 to 32.

**Comment by nekonomicon** [comment]
Try create game with maxplayer value over 32 twice.
It causes crash.
But may be needs fix for engine.

File: mainui/menu_creategame.cpp (hunk 1 above)
Comment (nekonomicon) [inline mainui/menu_creategame.cpp]: @a1batross, it seems crash into SV_DeactivateServer, but needs same check to be fix it.

Comment (a1batross) [inline mainui/menu_creategame.cpp]: @nekonomicon, try to start a server with 8 players and restart it with 16 players. Seems engine uses a new value of `maxplayers` in DeactivateServer.

**Pull request merged**